src/App.vue

```diff
@@ -1,40 +1,52 @@
 <template lang="pug">
 v-app.app
-  v-dialog(v-model="loggedOut", width="50%", persistent)
-    girder-auth(
-        :register="true",
-        :oauth="true",
-        :forgot-password-url="forgotPasswordUrl")
-  v-dialog(v-model="uploader", width="70%")
-    girder-upload(v-if="uploader",
-        :dest="uploadDest",
-        multiple="multiple",
-        @done="$refs.girderBrowser.refresh()")
-  girder-data-browser(ref="girderBrowser",
-      v-if="!loggedOut && location",
-      :location.sync="location",
-      @click:newitem="uploader = true")
+  v-container
+    h4.display-1.pb-3 Girder Vue Demo
+    v-dialog(v-model="loggedOut", full-width, max-width="600px")
+      girder-auth(
+          :register="true",
+          :oauth="true",
+          :forgot-password-url="forgotPasswordUrl")
+    v-dialog(v-model="uploader", full-width, max-width="800px")
+      girder-upload(v-if="uploader",
+          :dest="uploadDest",
+          multiple="multiple",
+          @done="$refs.girderBrowser.refresh(); uploader = false;")
+    v-dialog(v-model="newFolder", full-width, max-width="800px")
+      girder-upsert-folder(v-if="newFolder",
```

Comment:
I've seen this pattern in Vuetify, to not reset the contents of a dialog until a delay that accounts for the closing animation. Would this pattern help here?

https://vuetifyjs.com/en/components/data-tables#example-crud

```js
      close () {
        this.dialog = false
        setTimeout(() => {
          this.editedItem = Object.assign({}, this.defaultItem)
          this.editedIndex = -1
        }, 300)
      },
```

Not super pretty but there may not be a cleaner way.

Reply:
For the folder-creation dialog it would work perfectly.

For the upload dialog flash issue, I don't think this would be an appropriate fix.  The component resets itself after a successful upload, so we'd need to modify the main upload function with a callback to delay the reset or something like that.

Reply:
I personally think a resetted Upload with dialog dismiss animation feels smoother than suddenly disappeared dialog. If we still don't like the behavior, I think what Jeff said would work by wrapping just this [line](https://github.com/girder/girder_web_components/blob/fa23eb6723b495efa2d59752a2db411d49d87604/src/components/Upload.vue#L165) in a setTimeout()?
And we could do a similar thing to UpsertFolder, but if we don't want to do that, by changing the *v-if="newFolder"* to *:key="Math.random()"* would make it reset while keeping the dialog close animation. 
(I actually think we should keep the UpsertFolder state until the location has changed with *:key="location.id"*)

Reply:
@matthewma7 I added a state indicator, but for the life of me I cannot figure out how to keep the icons from bouncing from the height difference because of the height of the border.

I've tried `box-sizing: border-box` but this doesn't seem to have any effect on the buttons.   Any ideas?

Reply:
(I focused on the dialog close annimation and didn't aware you are aiming to to control the life cycle at first)
I actually learned about *key* when using React, but it turns out it is same for Vue, the documentation is rather hidden [here](https://vuejs.org/v2/api/#key).
*It can also be used to force replacement of an element/component instead of reusing it.*

Reply:
re: @matthewma7 @jeffbaumes 

#57 is my attempt at killing a few different birds with 1 stone.  if we merge that into this branch, would that solve our dismiss animation/state reset issues?